File: px-checkout/src/main/java/com/mercadopago/android/px/internal/repository/ChargeRepository.java
package com.mercadopago.android.px.internal.repository;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import com.mercadopago.android.px.model.commission.PaymentTypeChargeRule;
import java.math.BigDecimal;

public interface ChargeRepository {

    @NonNull
    BigDecimal getChargeAmount(@NonNull String paymentTypeId);

    @Nullable
    PaymentTypeChargeRule getChargeRule(@NonNull String paymentTypeId);
}

File: px-checkout/src/main/java/com/mercadopago/android/px/internal/repository/ChargeRepository.kt
package com.mercadopago.android.px.internal.repository

import com.mercadopago.android.px.model.commission.PaymentTypeChargeRule
import java.math.BigDecimal

//Should be internal but it's used in PaymentTypeChargeRule
interface ChargeRepository {
    val customCharges: List<PaymentTypeChargeRule>
    fun getChargeAmount(paymentTypeId: String): BigDecimal
    fun getChargeRule(paymentTypeId: String): PaymentTypeChargeRule?
    fun configure(customCharges: List<PaymentTypeChargeRule>)
    fun reset()
}